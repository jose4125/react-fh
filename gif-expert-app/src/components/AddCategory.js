import { useState } from "react";
import PropTypes from "prop-types";

export default function AddCategory({ onAddCategory }) {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim()) {
      onAddCategory(inputValue);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleChange} />
      <button type="submit">Add category</button>
    </form>
  );
}

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
};
