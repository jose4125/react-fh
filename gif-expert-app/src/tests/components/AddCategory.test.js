import React from "react";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("AddCategory tests", () => {
  let wrapper;
  const props = {
    onAddCategory: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<AddCategory {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change the input field", () => {
    const value = "goku";
    wrapper.find("input").simulate("change", {
      target: {
        value,
      },
    });
  });

  test("should not call onAddCategory", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(props.onAddCategory).not.toHaveBeenCalled();
  });

  test("should call onAddCategory", () => {
    const value = "goku";

    wrapper.find("input").simulate("change", {
      target: {
        value,
      },
    });

    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(props.onAddCategory).toHaveBeenCalled();
    expect(props.onAddCategory).toHaveBeenCalledWith(value);
    expect(props.onAddCategory).toHaveBeenCalledTimes(1);
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
