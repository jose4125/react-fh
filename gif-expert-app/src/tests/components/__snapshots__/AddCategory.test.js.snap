// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory tests should render properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Add category
  </button>
</form>
`;
